homeassistant/components/calendar/__init__.py

```diff
@@ -743,3 +753,22 @@ async def async_create_event(entity: CalendarEntity, call: ServiceCall) -> None:
         EVENT_END: end,
     }
     await entity.async_create_event(**params)
+
+
+async def async_list_events_service(
+    calendar: CalendarEntity, service_call: ServiceCall
+) -> dict[str, Any]:
+    """Handle snapshot services calls."""
+    if not service_call.return_values:
```

Comment:
I wonder if this is something we should allow specifying during service registration. It will also help us in the frontend to differentiate between return-only services

Reply:
That's a good idea. If we are going to require that you only ask for return values for services with return values then it makes sense to me.